src/pages/dogs/index.jsx

```diff
@@ -27,6 +29,15 @@ export default function DogsPage() {
   const [loading, setLoading] = useState(true);
 
   const [limitedAssociation, setLimitedAssociation] = useState(null);
+  const [userRole, setUserRole] = useState(null);
+
+  useEffect(() => {
```

Comment:
don't think useEffect is needed here, since this component gets re-rendered every time session changes from SessionManager